bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncDeviceOutletHandler.java

```diff
@@ -0,0 +1,238 @@
+                } catch (ParseException e) {
+                    logger.error("Could not parse timestamp: {}", e.getMessage());
+                }
+            } else {
+                logger.trace("Using cached response {}", responses);
+                String[] responseStrings = responses.split("/?");
+                responseStatus = responseStrings[0];
+                responseEnergyHistory = responseStrings[1];
+            }
+
+            if (responseStatus.equals(EMPTY_STRING) || responseEnergyHistory.equals(EMPTY_STRING)) {
+                return;
+            }
+
+            outletStatus = VeSyncConstants.GSON.fromJson(responseStatus, VeSyncV2BypassOutletStatus.class);
+            energyHistory = VeSyncConstants.GSON.fromJson(responseEnergyHistory, VeSyncV2BypassEnergyHistory.class);
+
+            if (outletStatus == null || energyHistory == null) {
+                return;
+            }
+            if (!cachedDataUsed) {
+                cachedResponse.putValue(responseStatus + "?" + responseEnergyHistory);
+            }
+        }
+
+        // Bail and update the status of the thing - it will be updated to online by the next search
+        // that detects it is online.
+        if (outletStatus.isMsgDeviceOffline()) {
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        } else if (outletStatus.isMsgSuccess()) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        if (!"0".equals(outletStatus.getCode())) {
+            logger.warn("Check Thing type has been set - API gave a unexpected response for an Outlet");
+            return;
+        }
+
+        updateState(DEVICE_CHANNEL_ENABLED,
+                OnOffType.from(MODE_ON.equals(outletStatus.outletResult.result.getDeviceStatus())));
+        updateState(DEVICE_CHANNEL_CURRENT, new QuantityType<>(outletStatus.outletResult.result.current, Units.AMPERE));
+        updateState(DEVICE_CHANNEL_VOLTAGE, new QuantityType<>(outletStatus.outletResult.result.voltage, Units.VOLT));
+        updateState(DEVICE_CHANNEL_ENERGY, new QuantityType<>(outletStatus.outletResult.result.energy, Units.WATT));
+        updateState(DEVICE_CHANNEL_POWER,
+                new QuantityType<>(outletStatus.outletResult.result.power, MetricPrefix.KILO(Units.WATT_HOUR)));
+        updateState(DEVICE_CHANNEL_HIGHEST_VOLTAGE,
+                new QuantityType<>(outletStatus.outletResult.result.highestVoltage, Units.VOLT));
+        updateState(DEVICE_CHANNEL_VOLTAGE_PT_STATUS, OnOffType.from(outletStatus.outletResult.result.voltagePTStatus));
+        updateState(DEVICE_CHANNEL_ENERGY_WEEK, new DecimalType(getEnergy(energyHistory, 7)));
+        updateState(DEVICE_CHANNEL_ENERGY_MONTH, new DecimalType(getEnergy(energyHistory, 30)));
+    }
+
+    private static long getTimestampForToday() throws ParseException {
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
I suspect there may be a simpler way to get this with openhabs lib's, however I'll have a look through the API's in a min, and have a play. I think you want 02:00 in UTC of the current day - is that correct? or are you 2 hours offset from UTC, and that's the reason for the 02:00 do you know? So its actually 00:00 at UTC this should be requesting?

Reply:
> If I remember correctly, I found the value via try and error. Because I was getting an offset of the actual time when retrieving the values. I’m -0100 from UTC so this is maybe something else. Maybe the location of their servers? Unfortunately I didn’t get any feedback from other users.

Reply:
Sounds good, lets keep it as is then. If anyone does raise an issue it was developed against -01:00UTC, so it can be recalculated at that point, based on their time-zone if required.